src/arkode/arkode_ls.c

```diff
@@ -2547,7 +2547,8 @@ int arkLsMPSolve(void* arkode_mem, N_Vector r, N_Vector z, sunrealtype tol, int
   approximation routines.
   ---------------------------------------------------------------*/
 int arkLsDQJac(sunrealtype t, N_Vector y, N_Vector fy, SUNMatrix Jac,
-               void* arkode_mem, N_Vector tmp1, N_Vector tmp2, N_Vector tmp3)
+               void* arkode_mem, N_Vector tmp1, N_Vector tmp2,
+               SUNDIALS_MAYBE_UNUSED N_Vector tmp3)
```

Comment:
> Are we certain that no compiler will take issue with user-supplied functions that mark different "unused" arguments from the `typedef`?

The unused attribute is just to suppress compiler warnings it does not have an impact on the compatibility of a function with a function pointer `typedef`. 

> Alternately, should the `typedef in (*ARKLsJacFn)` definition (as well as the definition of all other user-supplied functions) be updated to specify all of the potentially-unused arguments?

I don't think it makes sens to use the attribute in the `typedef` as this might hide warnings (I'm not sure if it will) that a user might want to see.